### PR TITLE
feat(cli): add support for minor releases

### PR DESCRIPTION
**Pull request by joshblack**
As part of our endgame, this PR adds support for automating `minor` releases from `@carbon/cli`.

#### Changelog

**New**

**Changed**

- Clean-up release command and refactor helpers into functions
- Update release command to support a `minor` release bump
- Pull out logger into dedicated file to be used by multiple commands

**Removed**

### CODE REVIEW

File: packages/cli/src/commands/release.js [hunk 1]
```diff
@@ -71,15 +73,98 @@ async function release({ bump }) {
...
+  try {
+    // This command will fail is no upstream is present, with `catch` we can
+    // create the appropriate remote before running the next commands
+    await execa('git', ['remote', 'get-url', 'upstream']);
+  } catch {
+    await execa('git', [
+      'remote',
+      'add',
+      'upstream',
+      'git@github.com:carbon-design-system/carbon.git',
```
Comment (emyarod) [inline packages/cli/src/commands/release.js:149]: just curious, is there a reason for using the ssh url over the https url?

**Review by emyarod — APPROVED** [review]
just one question, otherwise looks good to me!

File: packages/cli/src/commands/release.js (hunk 1 above)
Comment (joshblack) [reply packages/cli/src/commands/release.js:149]: Ah fair, didn't think about the difference. This is the URL I use by default which is the only reason lol. Would it be helpful to add a prompt to ask which one to use?

Comment (emyarod) [reply packages/cli/src/commands/release.js:149]: it's probably fine to just use one or the other, I just noticed that GitHub only exposes the https url on repo pages now

**Review by aledavila — APPROVED** [review]
other than andrew's concern. looks good

**Review by jnm2377 — APPROVED** [review]
Awesome!!!

**Pull request merged**